client/test/webview/TutorialGalleryView.test.ts

```diff
@@ -0,0 +1,165 @@
+        createWebviewPanelStub.should.have.been.called;
+    });
+
+    it('should reveal tutorial gallery page if already open', async () => {
+
+        const tutorialGalleryView: TutorialGalleryView = new TutorialGalleryView(context);
+        await tutorialGalleryView.openView(true);
+        await tutorialGalleryView.openView(true);
+
+        createWebviewPanelStub.should.have.been.calledOnce;
+
+        should.equal(createWebviewPanelStub.getCall(1), null);
+    });
+
+    it('should dispose tutorial gallery page', async () => {
+        const filterSpy: sinon.SinonSpy = mySandBox.spy(Array.prototype, 'filter');
+
+        createWebviewPanelStub.returns({
+            webview: {
+                onDidReceiveMessage: mySandBox.stub()
+            },
+            reveal: (): void => {
+                return;
+            },
+            title: 'Tutorial Gallery',
+            onDidDispose: mySandBox.stub().yields(),
+            onDidChangeViewState: mySandBox.stub()
+        });
+
+        const tutorialGalleryView: TutorialGalleryView = new TutorialGalleryView(context);
+
+        mySandBox.stub(tutorialGalleryView, 'getHTMLString').resolves('<html>Tutorial Gallery</html>');
+        await tutorialGalleryView.openView(true);
+
+        createWebviewPanelStub.should.have.been.calledWith(
+            'tutorialGallery',
+            'Tutorial Gallery',
+            vscode.ViewColumn.One,
+            {
+                enableScripts: true,
+                retainContextWhenHidden: true,
+                enableCommandUris: true,
+                localResourceRoots: [
+                    vscode.Uri.file(path.join(context.extensionPath, 'resources'))
+                ]
+            }
+        );
+
+        createWebviewPanelStub.should.have.been.calledOnce;
+
+        filterSpy.getCall(1).thisValue[filterSpy.getCall(1).thisValue.length - 1].webview.html.should.equal('<html>Tutorial Gallery</html>');
+    });
+
+    it('should send telemetry event on openPanelInner', async () => {
+        const panel: vscode.WebviewPanel = {
+            title: 'Tutorial Gallery'
+        } as vscode.WebviewPanel;
+
+        const tutorialGalleryView: TutorialGalleryView = new TutorialGalleryView(context);
+        await tutorialGalleryView.openPanelInner(panel);
+
+        reporterStub.should.have.been.calledWith('openedView', {name: 'Tutorial Gallery'});
+
+    });
+
+    it('should return string on getHTMLString', async () => {
+        const extensionPath: any = ExtensionUtil.getExtensionPath();
+        const tutorialsPath: string = path.join(extensionPath, 'tutorials.json');
+        const json: any = await fs.readJson(tutorialsPath);
+        const allSeries: any = json.series;
+
+        const tutorialView: TutorialGalleryView = new TutorialGalleryView(context);
+
+        const getTutorialGalleryPage: sinon.SinonStub = mySandBox.stub(TutorialGalleryView.prototype, 'getTutorialGalleryPage').resolves('<html>Tutorial Gallery</html>');
```

Comment:
why are you stubbing this, you want it to get the actual html otherwise it's not testing anything